lib/internal/process/esm_loader.js

```diff
@@ -61,10 +33,20 @@ exports.setup = function() {
   })();
   loaderResolve(loaderPromise);
 
-  setImportModuleDynamicallyCallback(async (referrer, specifier) => {
-    const loader = await loaderPromise;
-    return loader.import(specifier, normalizeReferrerURL(referrer));
-  });
-
   exports.ESMLoader = ESMLoader;
 };
+
+const config = process.binding('config');
+if (config.experimentalModules || config.experimentalVMModules) {
+  setInitializeImportMetaObjectCallback((wrap, meta) => {
+    if (wrap.initializeImportMeta !== undefined)
+      wrap.initializeImportMeta(meta);
+  });
+
+  setImportModuleDynamicallyCallback(async (referrer, specifier, wrap) => {
+    if (wrap !== undefined && wrap.importModuleDynamically !== undefined)
+      return wrap.importModuleDynamically(specifier);
```

Comment:
Are functions checked for being functions somewhere? Like `wrap.importModuleDynamically` and `wrap.initializeImportMeta` are checked for not being `undefined` but not for being functions.

Reply:
users can't touch ModuleWrap instances and even if they could they're frozen. I'll look into scripts when I have a free moment.